src/app/modules/group/components/pending-request/pending-request.component.ts

```diff
@@ -4,160 +4,40 @@ import {
   Input,
   OnChanges,
   SimpleChanges,
-  OnDestroy,
+  ContentChild, TemplateRef, Output, EventEmitter
 } from '@angular/core';
 import { SortEvent } from 'primeng/api/sortevent';
-import { MessageService } from 'primeng/api';
-import {
-  ERROR_MESSAGE,
-} from '../../../../shared/constants/api';
-import { TOAST_LENGTH } from '../../../../shared/constants/global';
-import { Observable, forkJoin, Subject, merge, of } from 'rxjs';
-import { GetRequestsService, PendingRequest } from '../../http-services/get-requests.service';
-import { Action, RequestActionsService } from '../../http-services/request-actions.service';
 import { GridColumn, GridColumnGroup } from '../../../shared-components/components/grid/grid.component';
-import { map, switchMap } from 'rxjs/operators';
-import { fetchingState, isReady, readyState } from 'src/app/shared/helpers/state';
-
-
-
-interface Result {
-  countRequests: number;
-  countSuccess: number;
-}
-
-const groupColumn = { field: 'group.name', header: 'GROUP' };
+import { Action } from 'src/app/modules/group/http-services/request-actions.service';
 
 @Component({
   selector: 'alg-pending-request',
   templateUrl: './pending-request.component.html',
   styleUrls: [ './pending-request.component.scss' ],
 })
-export class PendingRequestComponent implements OnInit, OnChanges, OnDestroy {
-  @Input() groupId?: string;
-  @Input() showSwitch = true;
+export class PendingRequestComponent<T> implements OnInit, OnChanges {
 
-  columns: GridColumn[] = [
-    { field: 'user.login', header: 'USER' },
-    { field: 'at', header: 'REQUESTED ON' },
-  ];
-  subgroupSwitchItems = [
-    { label: 'This group only', includeSubgroup: false },
-    { label: 'All subgroups', includeSubgroup: true }
-  ];
-  requests: PendingRequest[] = [];
-  selection: PendingRequest[] = [];
-  panel: GridColumnGroup[] = [];
-  currentSort: string[] = [];
-  includeSubgroup = false;
-  collapsed = true;
+  @Input() columns: GridColumn[] = [];
+  @Input() requests: T[] = [];
+  @Input() state: 'fetching' | 'processing' | 'ready' | 'fetchingError' = 'fetching';
 
-  state: 'fetching' | 'accepting' | 'rejecting' | 'ready' |'fetchingError' = 'fetching';
+  @Output() refetch = new EventEmitter<string[]>();
```

Comment:
If it is only used to sort, call it "sort", it is difficult to understand what is this list of string otherwise.

Reply:
fixed